src/manage.c

```diff
@@ -1528,13 +1529,58 @@ fork_osp_scan_handler (task_t task, target_t target, int from,
 
   setproctitle ("OSP: Handling scan %s", report_id);
 
-  rc = handle_osp_scan (task, global_current_report, report_id);
+  rc = handle_osp_scan (task, global_current_report, report_id, 0);
   g_free (report_id);
   rc = handle_osp_scan_end (task, rc);
   gvm_close_sentry ();
   exit (rc);
 }
 
+/**
+ * @brief Prepare an OSP scan and add it to the gvmd scan queue.
+ *
+ * @param[in]   task       The task.
+ * @param[in]   start_from 0 start from beginning, 1 continue from stopped,
+ *                         2 continue if stopped else start from beginning.
+ * @param[out]  report_id_return   UUID of the report.
+ *
+ * @return 0 on success, -1 on failure.
+ */
+static int
+queue_osp_task (task_t task, int start_from, char **report_id_return)
+{
+  char *report_id = NULL;
+  report_t report = 0;
+
+  if (report_id_return)
+    *report_id_return = NULL;
+
+  if (run_osp_scan_get_report (task, start_from, &report_id))
+    return -1;
+
+  if (find_resource_no_acl ("report", report_id, &report))
+    {
+      g_warning ("%s: error getting report '%s'",
+                 __func__, report_id);
+      g_free (report_id);
+      return -1;
+    }
+  else if (report == 0)
+    {
+      g_warning ("%s: could not find report '%s'",
+                 __func__, report_id);
+      g_free (report_id);
+      return -1;
+    }
+
+  scan_queue_add (report);
+  set_task_run_status (task, TASK_STATUS_REQUESTED);
+  set_report_scan_run_status (report, TASK_STATUS_REQUESTED);
+  g_debug ("%s: report %s (%llu) added to scan queue",
+           __func__, report_id, report);
+  return 0;
```

Comment:
`report_id` could be freed here.
```suggestion
  g_free (report_id);
  return 0;
```